src/kw/kw.cpp

```diff
@@ -230,37 +229,14 @@ string KanoWorld::get_refresh_header(const string& token) {
 
 
 bool KanoWorld::is_logged_in(const bool verbose) {
-    std::time_t now = std::time(nullptr);
-
-    if (!load_data()) {
-        return false;
-    }
-
-    std::time_t duration = atol(this->get_expiration_date().c_str());
-
-    if (!duration) {
-        return false;
-    }
-
-    double seconds = difftime(now, duration);
-
-    if (verbose) {
-        char now_str[26];
-        ctime_r(static_cast<const time_t*>(&now), now_str);
-
-        char duration_str[26];
-        ctime_r(static_cast<const time_t*>(&duration), duration_str);
-
-        cout << "--- Am_I_Logged_In() requested - Time: " << now << " - "
-             << now_str << endl;
+    return !this->is_token_expired();
+}
 
-        cout << "--- Token expires: " << duration << " - "
-             << duration_str << endl;
-        cout << "--- Difference in seconds: " << seconds << endl;
-        cout << "--- Token valid? " << (seconds < 0 ? "Yes" : "No") << endl;
-    }
 
-    return seconds < 0;
+bool KanoWorld::is_token_expired() const {
+    return
+        duration_cast<milliseconds>(system_clock::now().time_since_epoch()) >
+        this->get_expiration_date();
```

Comment:
Do we want to have some sanity checking in case the clock hasn't been set?

Reply:
Let's test it the way it is and spin off bug tickets if any arise 🤔